src/rhsmlib/facts/host_collector.py

```diff
@@ -67,7 +67,10 @@ def get_all(self):
         host_facts.update(firmware_info_dict)
 
         locale_info = {}
-        locale_info['system.default_locale'] = ".".join(locale.getdefaultlocale())
+        effective_locale = 'Unknown'
+        if locale.getdefaultlocale()[0] is not None:
```

Comment:
Is it not possible for locale.getdefaultlocale() to return ("Something_not_none", None)?